### ISSUE TITLE
[Bug]: Automatic Suggestions "Search for Share Recipients" Not Recommending Based on Frequently Shared Users 

### ISSUE THREAD

**Issue by alperozturk96** [nextcloud/server#45752]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When user wants to share any file or folder from Files -> Sharing user recommendations are not based frequently shared users.

### Steps to reproduce

1. Share files with one user multiple times and have multiple users in your instance
2. During new shares, user have to explicitly enter user's name

### Expected behavior

If a user shared files with one or more users multiple times, share recommendations must be based on frequently shared users.

### Installation method

None

### Nextcloud Server version

27

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by sorbaugh** [comment]
@Fenn-CS adding you for a first estimation, wdyt?

**Comment by ChristophWurst** [comment]
@Fenn-CS friendly ping

**Comment by Fenn-CS** [comment]
@Fenn-CS friendly ping

> Thanks for the ping Christoph 

Few things:

- This is definitely something that the server api would supply to the UI
- Since I don't know about this "feature" before now, I need to figure out IF this was  feature that broke or if it's something entirely new we have to implement and this would guide the estimations. 

@artonge Is this how the share recommendation api worked (pre nc27)? 

cc: @Altahrim

**Comment by Fenn-CS** [comment]
So after testing with @Altahrim it is a thing but it works as expected on 27 right up to master. I suspect that the recommendation settings for the tester that saw this problem could be the limiting factor here;

**Comment by alperozturk96** [comment]
I ran the command `docker pull nextcloud:27.1.11` to install a fresh Nextcloud instance.

I added three users: Tyler, Jack, and Terence.

When I shared an image using Tyler, the next time I wanted to share a file, Tyler was automatically recommended. This feature worked well, which wasn't the case with previous versions.

However, even after setting `'sharing.minSearchStringLength' => 10`, I still see share recommendations without typing.

[Doc](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#sharing-minsearchstringlength) for minSearchStringLenght

Demo


https://github.com/user-attachments/assets/51be50ef-bb5a-409d-8954-c17c5469c717
